kubernetes/config/kube_config.py

```diff
@@ -22,151 +22,257 @@
 from kubernetes.client import configuration
 from oauth2client.client import GoogleCredentials
 
-_temp_files = []
+from .incluster_config import ConfigException
+
+_temp_files = {}
 
 
 def _cleanup_temp_files():
-    for f in _temp_files:
-        os.remove(f)
+    for k in _temp_files:
+        os.remove(_temp_files[k])
 
 
 def _create_temp_file_with_content(content):
     if len(_temp_files) == 0:
```

Comment:
I believe even with one call, the temp file(s) will be removed. The if statement make sure we register cleanup handler once, on the first call.